pkg/antctl/raw/featuregates/command.go

```diff
@@ -0,0 +1,207 @@
+		Command.Long = "Get current Antrea agent feature gates info"
+	} else if runtime.Mode == runtime.ModeController && runtime.InPod {
+		Command.RunE = controllerLocalRunE
+		Command.Long = "Get current Antrea controller feature gates info."
+	} else if runtime.Mode == runtime.ModeController && !runtime.InPod {
+		Command.Long = "Get Antrea feature gates info including Controller and Agent"
+		Command.RunE = controllerRemoteRunE
+	}
+}
+
+func setupKubeconfig(kubeconfig *rest.Config) {
+	kubeconfig.Insecure = true
+	kubeconfig.CAFile = ""
+	kubeconfig.CAData = nil
+	kubeconfig.GroupVersion = &schema.GroupVersion{Group: "", Version: ""}
+	kubeconfig.NegotiatedSerializer = scheme.Codecs.WithoutConversion()
+	if runtime.InPod {
+		if runtime.Mode == runtime.ModeAgent {
+			kubeconfig.Host = net.JoinHostPort("127.0.0.1", strconv.Itoa(apis.AntreaAgentAPIPort))
+			kubeconfig.BearerTokenFile = agentapiserver.TokenPath
+		} else {
+			kubeconfig.Host = net.JoinHostPort("127.0.0.1", strconv.Itoa(apis.AntreaControllerAPIPort))
+			kubeconfig.BearerTokenFile = controllerapiserver.TokenPath
+		}
+	}
+}
+
+func agentRunE(cmd *cobra.Command, _ []string) error {
+	return featureGateRequest(cmd, runtime.ModeAgent)
+}
+
+func controllerLocalRunE(cmd *cobra.Command, _ []string) error {
+	return featureGateRequest(cmd, runtime.ModeController)
+}
+
+func controllerRemoteRunE(cmd *cobra.Command, _ []string) error {
+	return featureGateRequest(cmd, "remote")
+}
+
+func featureGateRequest(cmd *cobra.Command, mode string) error {
+	kubeconfigPath, err := cmd.Flags().GetString("kubeconfig")
+	if err != nil {
+		return err
+	}
+	kubeconfig, err := runtime.ResolveKubeconfig(kubeconfigPath)
+	if err != nil {
+		return err
+	}
+	restconfigTmpl := rest.CopyConfig(kubeconfig)
+	setupKubeconfig(restconfigTmpl)
+	if server, err := Command.Flags().GetString("server"); err != nil {
+		kubeconfig.Host = server
+	}
+
+	k8sClientset, err := kubernetes.NewForConfig(kubeconfig)
+	if err != nil {
+		return fmt.Errorf("failed to create K8s clientset: %w", err)
+	}
+	antreaClientset, err := antrea.NewForConfig(kubeconfig)
+	if err != nil {
+		return fmt.Errorf("error when creating Antrea clientset: %w", err)
+	}
+
+	var agentResp []featuregates.Response
+	var controllerResp []featuregates.Response
+	var agentClient *rest.RESTClient
+	var controllerClient *rest.RESTClient
+
+	switch mode {
+	case runtime.ModeAgent:
+		agentClient, err = rest.RESTClientFor(restconfigTmpl)
+	case runtime.ModeController:
+		controllerClient, err = rest.RESTClientFor(restconfigTmpl)
+
+	}
+	if err != nil {
+		return fmt.Errorf("fail to creating %s rest client %w", mode, err)
+	}
+
+	if mode == "remote" {
+		if agentClient, err = getAgentClient(k8sClientset, antreaClientset, restconfigTmpl); err != nil {
+			return err
+		}
+		if controllerClient, err = getControllerClient(k8sClientset, antreaClientset, restconfigTmpl); err != nil {
+			return err
+		}
+	}
+
+	if mode == runtime.ModeAgent || mode == "remote" {
```

Comment:
i think it would be better to test `agentClient` for `nil` here instead of this if condition. Same below for `controllerClient`.